# Importing Libraries

In [1]:
from bs4 import BeautifulSoup
import pandas as pd
import requests
import datetime
import smtplib #used for sending emails
import time
import csv
import re

# Connecting to the website

In [2]:
URL="https://www.amazon.co.uk/Data-Analyst-Gifts-T-Shirts-T-Shirt/dp/B08K5FPZ1Q/ref=sr_1_6?crid=OXVI6BX1P0JV&keywords=data+analyst+tshirt&qid=1646580310&sprefix=data+analyst+%2Caps%2C134&sr=8-6"

headers = {"User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36"}

page = requests.get(URL,headers=headers)

soup1 = BeautifulSoup(page.content, "html.parser")

soup2 = BeautifulSoup(soup1.prettify(), "html.parser")

title = soup2.find(id="productTitle").get_text()

price = soup2.find(id="corePriceDisplay_desktop_feature_div").get_text()

price.strip()
p = price.split()[0]

print(title)
print(p)


                   Data Analyst T-Shirt
                  
£15.90


In [3]:
#cleaning the obtained data
title=title.strip()
p = p.strip()[1:]
p = float(p)

print(title)
print(p)

Data Analyst T-Shirt
15.9


# Making a CSV file out of the data obtained

In [4]:
today = datetime.date.today()
header = ["Title","Price","Date"]
data = [title, p, today]

with open("AmazonWebScrapperDataset.csv","w", newline='', encoding="UTF8") as f:
    writer = csv.writer(f)
    writer.writerow(header)
    writer.writerow(data)

In [5]:
df = pd.read_csv(r"D:\New Projects\Amazon Web Scraping project\AmazonWebScrapperDataset.csv")

print(df)

                  Title  Price        Date
0  Data Analyst T-Shirt   15.9  2022-03-07


# Appending data to the existing CSV file 

In [6]:
with open("AmazonWebScrapperDataset.csv","a+", newline='', encoding="UTF8") as f:
    writer = csv.writer(f)
    writer.writerow(data)

# Making an Email Function to get updated when T-shirt ia less than £15.90

In [7]:
def send_mail():
    server = smtplib.SMTP_SSL('smtp.gmail.com',465)
    server.ehlo
    server.login('sendersemail@gmail.com','********')
    
    subject = "The Shirt you want is below $14! Now is your chance to buy!"
    body = "Shamika, This is the moment we have been waiting for. Now is your chance to pick up the shirt of your dreams. Don't mess it up! Link here: https://www.amazon.com/Funny-Data-Systems-Business-Analyst/dp/B07FNW9FGJ/ref=sr_1_3?dchild=1&keywords=data+analyst+tshirt&qid=1626655184&sr=8-3"
   
    msg = f"Subject: {subject}\n\n{body}"
    
    server.sendmail(
        'sendersemail@gmail.com','recievers@gmail.com',
        msg
     
    )

# Making a Check Price function

check_price funtion checks if the price is less than 15.90 and gives an alert on the email id with the link to buy at a low price.

In [8]:
def check_price():
    URL="https://www.amazon.co.uk/Data-Analyst-Gifts-T-Shirts-T-Shirt/dp/B08K5FPZ1Q/ref=sr_1_6?crid=OXVI6BX1P0JV&keywords=data+analyst+tshirt&qid=1646580310&sprefix=data+analyst+%2Caps%2C134&sr=8-6"

    headers = {"User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36"}

    page = requests.get(URL,headers=headers)

    soup1 = BeautifulSoup(page.content, "html.parser")

    soup2 = BeautifulSoup(soup1.prettify(), "html.parser")

    title = soup2.find(id="productTitle").get_text()

    price = soup2.find(id="corePriceDisplay_desktop_feature_div").get_text()




    #removing white spaces from the price_new1 string
    price.strip()
    p = price.split()[0]
    
    p=p.strip()[1:]
    p = float(p)
    title=title.strip()
    
    today = datetime.date.today()
    header = ["Title","Price","Date"]
    data = [title, p, today]
    
    with open("AmazonWebScrapperDataset.csv","a+", newline='', encoding="UTF8") as f:
        writer = csv.writer(f)
        writer.writerow(data)
        
    if(p < 15):
        
        send_mail()

In [ ]:
while(True):
    check_price()
    time.sleep(5)

# Checking the Updated CSV file 

In [10]:
df = pd.read_csv(r"D:\New Projects\Amazon Web Scraping project\AmazonWebScraperDataset.csv")

print(df)

                  Title  Price        Date
0  Data Analyst T-Shirt   15.9  2022-03-07
1  Data Analyst T-Shirt   15.9  2022-03-07
2  Data Analyst T-Shirt   15.9  2022-03-07
3  Data Analyst T-Shirt   15.9  2022-03-07
4  Data Analyst T-Shirt   15.9  2022-03-07
5  Data Analyst T-Shirt   15.9  2022-03-07
